src/helpers/ApiHelper.js

```diff
@@ -317,6 +326,45 @@ const ApiHelper = {
     // Skip over item since itemBarcode is not defined and checkoutProccessed is false
     logger.info(`Unable to sent POST request to checkin-service for Cancel Request Record (${item.id}); checkoutProccessed is false which indicates this item was not checked out`);
     return callback(null, item);
+  },
+  patchRecapItem (apiUrl, token, errorHandlerFn, item, callback) {
+    logger.info(`patching: apiUrl: ${apiUrl}, item: ${item}`)
+    if (item && typeof item === 'object' && item.id) {
+      //initialize the boolean flag to false until a successful post updates to true
+      item.recapProcessed = false;
+
+      if (item.checkoutProccessed === true && item.checkinProccessed === true) {
+        logger.info(`Posting Cancel Request Record (${item.id}) to Recap`);
+        const recapApiModel = {
+          success: true,
+          processed: true,
+        }
+
+        return axios.patch(`${apiUrl}/${item.id}`, recapApiModel, this.getApiHeaders(token))
+        .then(result => {
+          let processedItem = item;
+          processedItem.recapApiResponse = this.generateSuccessfulResponseObject(result);
+
+          if (this.isItemPostSuccessful(result, 'success')) {
+            logger.info(`Successfully posted Cancel Request Record (${item.id}) to Recap; assigned response to record`);
+            processedItem = Object.assign(processedItem, { recapProcessed: true}, { success: true});
+          }
+
+          return callback(null, processedItem);
+        })
+        .catch(error => {
+          const errorResponse = this.generateErrorResponseObject(error);
+          // Assign the error clean error object to the item
+          item.error = errorResponse;
+          // Handle retries or fatal errors by error status code
+          return errorHandlerFn(errorResponse, 'recap-service', item, callback);
+        })
+      }
+
+      // Skip item if checkoutProccessed or checkinProccessed is not true
+      logger.info(`Will not patch Cancel Request Record: (${item.id}) in Recap. checkoutProccessed is: ${item.checkoutProccessed} and checkinProccessed is ${item.checkinProccessed}`);
+      return callback(null, item);
+    }
```

Comment:
There sure is a lot of mixing of callbacks and promises in this app, which seems a shame, but does seem to be the style chosen.

Reply:
Totally agree